src/server.rs

```diff
@@ -1,209 +1,426 @@
-    #[allow(clippy::cast_precision_loss)]
-    let mul = 10u128.pow(dec.try_into()?) as f64;
+async fn order(
+    state: extract::State<Arc<State>>,
+    matched_path: MatchedPath,
+    path_result: Result<RawPathParams, RawPathParamsRejection>,
+    query: Query<HashMap<String, String>>,
+) -> Response {
+    match process_order(state, &matched_path, path_result, &query).await {
+        Ok((order_status, order_success)) => match order_success {
+            OrderSuccess::Created => (StatusCode::CREATED, Json(order_status)),
+            OrderSuccess::Found => (StatusCode::OK, Json(order_status)),
+        }
+        .into_response(),
+        Err(error) => match error {
+            OrderError::LessThanExistentialDeposit(existential_deposit) => (
+                StatusCode::BAD_REQUEST,
+                Json([InvalidParameter {
+                    parameter: AMOUNT.into(),
+                    message: format!("provided amount is less than the currency's existential deposit ({existential_deposit})"),
+                }]),
+            )
+                .into_response(),
+            OrderError::UnknownCurrency => (
+                StatusCode::BAD_REQUEST,
+                Json([InvalidParameter {
+                    parameter: CURRENCY.into(),
+                    message: "provided currency isn't supported".into(),
+                }]),
+            )
+                .into_response(),
+            OrderError::MissingParameter(parameter) => (
+                StatusCode::BAD_REQUEST,
+                Json([InvalidParameter {
+                    parameter,
+                    message: "parameter wasn't found".into(),
+                }]),
+            )
+                .into_response(),
+            OrderError::InvalidParameter(parameter) => (
+                StatusCode::BAD_REQUEST,
+                Json([InvalidParameter {
+                    parameter,
+                    message: "parameter's format is invalid".into(),
+                }]),
+            )
+                .into_response(),
+            OrderError::AlreadyProcessed(order_status) => {
+                (StatusCode::CONFLICT, Json(order_status)).into_response()
+            }
+            OrderError::InternalError => StatusCode::INTERNAL_SERVER_ERROR.into_response(),
+        },
+    }
+}
 
-    Ok(numfl / mul)
+async fn status(
+    state: extract::State<Arc<State>>,
+) -> ([(HeaderName, &'static str); 1], Json<ServerStatus>) {
+    (
+        [(header::CACHE_CONTROL, "no-store")],
+        ServerStatus {
+            description: ServerInfo {
+                version: env!("CARGO_PKG_VERSION"),
+                instance_id: String::new(),
+                debug: state.0.debug,
+                kalatori_remark: state.0.remark.clone(),
+            },
+            supported_currencies: vec![CurrencyInfo {
+                currency: "USDC".into(),
+                chain_name: "assethub-polkadot".into(),
+                kind: TokenKind::Assets,
```

Comment:
This yields `assets` instead of a singular `asset`, which violates the API defined in https://alzymologist.github.io/kalatori-api/